cpp/ql/src/semmle/code/cpp/models/implementations/StdMap.qll

```diff
@@ -51,6 +51,51 @@ class StdMapInsert extends TaintFunction {
   }
 }
 
+/**
+ * The standard map `emplace` and `emplace_hint` functions.
+ */
+class StdMapEmplace extends TaintFunction {
+  StdMapEmplace() {
+    this.hasQualifiedName("std", ["map", "unordered_map"], ["emplace", "emplace_hint"])
+  }
+
+  override predicate hasTaintFlow(FunctionInput input, FunctionOutput output) {
+    // flow from any parameter to qualifier and return value
+    // (here we assume taint flow from any constructor parameter to the constructed object)
+    // (where the return value is a pair, this should really flow just to the first part of it)
+    input.isParameterDeref(_) and
```

Comment:
I thought the `_` here, though something of an anti-pattern, was necessary to support varargs parameters (similarly to `FormattingFunction::hasTaintFlow`).  I'm not so sure now as the function actually uses variadic template parameters (or whatever you call `template<class... Args> ...`) not varargs, so each instantiation ought to have a fixed number of parameters (I think).

Reply:
I've made the change.  There are a few differences in `localTaint.expected` because the old version was matching parameter `-1` it seems. :S